### PR TITLE
Cache page size in ba_os_get_page_size()

### PR DESCRIPTION
**Pull request by ldorau**

### CODE REVIEW

File: src/base_alloc/base_alloc_linux.c [hunk 1]
```diff
@@ -33,4 +33,12 @@ void ba_os_free(void *ptr, size_t size) {
     (void)ret; // unused
 }
 
-size_t ba_os_get_page_size(void) { return sysconf(_SC_PAGE_SIZE); }
+size_t ba_os_get_page_size(void) {
+    static size_t Page_size = 0;
+
+    if (!Page_size) {
```
Comment (igchor) [inline src/base_alloc/base_alloc_linux.c:39]: Shouldn't we use util_init_once or something like that? This function can be called from multiple threads, right?

Comment (ldorau) [reply src/base_alloc/base_alloc_linux.c:39]: Right. So it is better to make it a TLS variable - no MT synchronization is needed then.
Done.

Comment (pbalcer) [reply src/base_alloc/base_alloc_linux.c]: using tls isn't free, especially for a library. For something this simple, util_init_once is the right thing to use...

Comment (ldorau) [reply src/base_alloc/base_alloc_linux.c]: OK, done

**Review by bratpiorka — APPROVED** [review]

**Review by kswiecicki — APPROVED** [review]

**Review by igchor — APPROVED** [review]

**Pull request merged**